kernel/cClosure.ml

```diff
@@ -399,14 +399,16 @@ end = struct
         let cb = lookup_constant cst env in
         shortcut_irrelevant info (UVars.subst_instance_relevance u cb.const_relevance);
         let ts = RedFlags.red_transparent info.i_flags in
-        if TransparentState.is_transparent_constant ts cst then begin
-          match Cmap_env.find_opt cst env.symb_pats with
-          | Some r ->
-            let b = match [@ocaml.warning "-4"] cb.const_body with Symbol b -> b | _ -> assert false in
-            raise (NotEvaluableConst (HasRules (u, b, r)))
-          | None -> ();
+        if TransparentState.is_transparent_constant ts cst then match cb.const_body with
+        | Undef _ | Def _ | OpaqueDef _ | Primitive _ ->
           Def (constant_value_in u cb.const_body)
-        end else
+        | Symbol b ->
+          let r = match Cmap_env.find_opt cst env.symb_pats with
```

Comment:
why not use `get` instead of having a few extra lines that just assert false

Reply:
The problem is that Option.get raises an exception. It should assert false instead, but this cannot go into such a simple PR. It's not the first time I want to have an exception-free Option.get, we should do something about it but tracking exception raising is annoying.

Reply:
I mean Cmap_env.get which does assert false

Reply:
Oh, yes, indeed.